### ISSUE TITLE
Master is failing with TensorFlow 2.0

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#17797]
Example from https://travis-ci.org/sympy/sympy/jobs/602486430?utm_medium=notification&utm_source=github_status

```
2019-10-24 19:57:46.217079: I tensorflow/core/platform/cpu_feature_guard.cc:145] This TensorFlow binary is optimized with Intel(R) MKL-DNN to use the following CPU instructions in performance critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA

To enable them in non-MKL-DNN operations, rebuild TensorFlow with the appropriate compiler flags.

2019-10-24 19:57:46.225175: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2300000000 Hz

2019-10-24 19:57:46.226008: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55ddb611e280 executing computations on platform Host. Devices:

2019-10-24 19:57:46.226086: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version

2019-10-24 19:57:46.226402: I tensorflow/core/common_runtime/process_util.cc:115] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.

Esympy/printing/tests/test_theanocode.py[38] ................/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/theano/gof/opt.py:251: UserWarning: WARNING: Supervisor is not added. Please build a FunctionGraphvia theano.compile.function_module.std_graph()or add the Supervisor class manually.

  sub_prof = optimizer.optimize(fgraph)

...................

...                                                                         [OK]

sympy/printing/tests/test_tree.py[1] .                                      [OK]

sympy/vector/tests/test_printing.py[4] ....                                 [OK]

________________________________________________________________________________

_________ sympy/printing/tests/test_tableform.py:test_TableForm_latex __________

Traceback (most recent call last):

  File "sympy/core/compatibility.py", line 144, in exec_

    exec("exec _code_ in _globs_, _locs_")

  File "<string>", line 1, in <module>

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/sympy-1.5.dev0-py2.7.egg/sympy/printing/tests/test_tensorflow.py", line 28, in <module>

    session = tf.Session()

AttributeError: 'module' object has no attribute 'Session'

========= tests finished: 1008 passed, 1 exceptions, in 26.06 seconds ==========

DO *NOT* COMMIT!

Traceback (most recent call last):

  File "<stdin>", line 10, in <module>

Exception: Tests failed
```

**Comment by moorepants** [comment]
We should set tensorflow < 2.0 in the travis.yml.

**Comment by asmeurer** [comment]
I had thought we added support for tensorflow 2.0. Maybe it was just a pre-release and they changed stuff since then.

**Comment by asmeurer** [comment]
See https://github.com/sympy/sympy/issues/17149 and the associated PR. I guess it was never finished. We should see if we can finish it up before the release.

**Comment by oscarbenjamin** [comment]
Closing as fixed in #17103
